// バージョンの定義
syntax = "proto3";
// パッケージの定義(名前の衝突を防ぐためのもの)
package updater_recipe;

service UpdaterRecipeService {
    rpc GetPackages(Empty) returns (GetPackagesReply);
    rpc GetFileInPackage(GetFileInPackageRequest) returns (GetFileInPackageReply);
}

// 引数なし用
message Empty {
}

message Package {
    string package_name = 1;
}

// サービス用に特化したmessage定義
message GetPackagesReply {
    repeated Package packages = 1;
}

message GetFileInPackageRequest {
    string package_name = 1;
}

message GetFileInPackageReply {
    repeated string file_full_path = 1;
}
